### PR TITLE
Shape and Image panels

### PR DESCRIPTION
**Pull request by nicolaskruchten**
Roughed in, lots of fit'n'finish needed :(

### CODE REVIEW

File: src/PlotlyEditor.js [hunk 1]
```diff
@@ -145,6 +145,36 @@ class PlotlyEditor extends Component {
...
+            this.props.afterDeleteShape(payload);
+          }
+          if (this.props.onUpdate) {
+            this.props.onUpdate();
+          }
+        }
+        break;
+
+      case EDITOR_ACTIONS.DELETE_IMAGE:
+        if (isNumeric(payload.imageIndex)) {
```
Comment (VeraZab) [inline src/PlotlyEditor.js]: I guess on a later pass we could combine these delete 'annotations' 'shapes' 'images' actions into one

File: src/components/containers/AnnotationAccordion.js [hunk 2]
```diff
@@ -35,7 +35,7 @@ class AnnotationAccordion extends Component {
         }
 
         const key = `annotations[${annotationIndex}]`;
-        const value = {text: 'new text'};
+        const value = {text: _('new text')};
```
Comment (VeraZab) [inline src/components/containers/AnnotationAccordion.js]: mercii

File: src/PlotlyEditor.js (hunk 1 above)
Comment (nicolaskruchten) [reply src/PlotlyEditor.js]: yeah, this was a dirty copy-paste job... We'll see if it turns out that these evolve in different directions and if not, we can DRY them up.

File: src/lib/index.js [hunk 3]
```diff
@@ -49,6 +51,8 @@ export {
   capitalize,
   clamp,
   connectAnnotationToLayout,
+  connectShapeToLayout,
+  connectImageToLayout,
   connectAxesToLayout,
   connectLayoutToPlot,
   connectToContainer,
```
Comment (VeraZab) [inline src/lib/index.js]: can we add these to the README

File: src/index.js [hunk 4]
```diff
@@ -60,13 +65,18 @@ import {
   StyleLayoutPanel,
   StyleLegendPanel,
   StyleNotesPanel,
+  StyleShapesPanel,
+  StyleImagesPanel,
   StyleTracesPanel,
 } from './default_panels';
```
Comment (VeraZab) [inline src/index.js]: same, can we add these to the README

File: src/lib/index.js (hunk 3 above)
Comment (nicolaskruchten) [reply src/lib/index.js]: yep

File: src/index.js (hunk 4 above)
Comment (nicolaskruchten) [reply src/index.js]: Hmmm well none of the other default panels are in the readme, so I would vote no...

Comment (VeraZab) [reply src/index.js]: I mean the positionningRef, the Accordions, the connect functions : )

File: src/default_panels/StyleImagesPanel.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+const StyleImagesPanel = ({localize: _}) => (
+  <ImageAccordion canAdd>
+    <Radio
+      attr="visible"
+      options={[
+        {label: _('Show'), value: true},
+        {label: _('Hide'), value: false},
+      ]}
+    />
+    <TextEditor attr="source" label={_('Source')} show />
```
Comment (VeraZab) [inline src/default_panels/StyleImagesPanel.js]: yeah, we need the component hein..
would you mind creating the issue so we don't forget..

Comment (nicolaskruchten) [reply src/default_panels/StyleImagesPanel.js]: yep. we can use `react-dropzone` I think, I used it for https://react-pivottable.js.org/ and it works great.

Comment (nicolaskruchten) [reply src/default_panels/StyleImagesPanel.js]: https://github.com/plotly/react-plotly.js-editor/issues/333

**Comment by nicolaskruchten** [comment]
PR feedback addressed!

**Comment by VeraZab** [comment]
💃

**Pull request merged**